### PR TITLE
Add list collector summary definition to schema

### PR DESCRIPTION
**Pull request by JamesGardiner**
Adds a `summary` definition in the list collector json.

### CODE REVIEW

**Comment by JamesGardiner** [comment]
[Refactor list collector summary to be defined in schema](https://trello.com/c/8u0SzPws/3135-refactor-list-collector-summary-to-be-defined-in-schema)

File: schemas/blocks/list_collector.json [hunk 1]
```diff
@@ -18,6 +18,35 @@
     "required": ["id", "value"],
     "additionalProperties": false
   },
+  "summary": {
+    "type": "object",
+    "properties":{
+      "type": {
+          "type": "string",
+          "enum": ["ListCollectorSummary"]
```
Comment (MebinAbraham) [inline schemas/blocks/list_collector.json]: Slightly confused on why we have a type for a 'Summary' that is defined within the list collector itself?. What other types would there be?

File: schemas/blocks/list_collector.json [hunk 2]
```diff
@@ -18,6 +18,35 @@
...
+              "description": "Summary text definitions using placeholders"
+          }
+        },
+        "required": [
+          "text",
+          "placeholders"
+        ]
+      }
+    },
+    "required": [
```
Comment (MebinAbraham) [inline schemas/blocks/list_collector.json]: We should add `"additionalProperties": false`

File: schemas/blocks/list_collector.json [hunk 3]
```diff
@@ -18,6 +18,35 @@
...
     "additionalProperties": false
   },
+  "summary": {
+    "type": "object",
+    "properties":{
+      "type": {
+          "type": "string",
+          "enum": ["ListCollectorSummary"]
+      },
+      "item_title": {
```
Comment (MebinAbraham) [inline schemas/blocks/list_collector.json]: For `item_title` we should be able to reuse the `#/string_with_placeholders"` definition?

https://github.com/ONSdigital/eq-schema-validator/blob/refactor-list-collector-summary/schemas/string_interpolation/definitions.json#L3-L26

File: schemas/blocks/list_collector.json (hunk 1 above)
Comment (JamesGardiner) [reply schemas/blocks/list_collector.json]: You're right, this isn't necessary

File: schemas/blocks/list_collector.json [hunk 4]
```diff
@@ -18,6 +18,18 @@
     "required": ["id", "value"],
     "additionalProperties": false
   },
+  "summary": {
```
Comment (MebinAbraham) [inline schemas/blocks/list_collector.json]: Should we make this key `required`?

File: tests/schemas/valid/test_list_collector.json [hunk 5]
```diff
@@ -200,6 +200,27 @@
               }]
             }]
           }
+        },
+        "summary": {
+          "item_titles": {
```
Comment (MebinAbraham) [inline tests/schemas/valid/test_list_collector.json]: Should this not be singular?. An item has a title?

File: schemas/blocks/list_collector.json (hunk 4 above)
Comment (JamesGardiner) [reply schemas/blocks/list_collector.json]: Asked @ajmaddaford about this yesterday - the consensus was it shouldn't be mandatory

**Pull request merged**